### PR TITLE
ec2_elb_facts: Handle AWS throttling when requesting ELB information

### PR DESCRIPTION
**Pull request by bobobox**
##### SUMMARY
Fixes issue where module would fail immediately upon encountering AWS API throttling.
Also makes minor changes to error handling.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_elb_facts.py

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides
```

##### ADDITIONAL INFORMATION
Uses the `AWSRetry.backoff` method both as a decorator and, where more efficient, directly as a wrapper function to  handle AWS API throttling for boto methods invoked in the module. 

Various instances of `BotoServerError` exceptions being caught and handled simply by invoking the  `fail_json` method have been removed, and replaced with a similar try/catch block in the `main` function. I'm not sure that this exception handling was particularly useful in the first place (and I'd be open to removing it completely), but moving it into `main` at least provides pretty much equivalent behavior without interfering with use of the `AWSRetry` decorator, which can handle specific cases of `BotoServerError`.

Also, removes questionable 'catch-all' clause on line 111 which would return an empty dict for tags.

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: Please import what you need rather than `*`

http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#python-imports

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by gundalow** [comment]
If this can get reviewed by the maintainers within the week we can look at getting this into the 2.3 release.

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (gundalow) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: Ah, I see you've just moved the code to satisfy CI, rather than these been new imports. If you could fix this it would be great.

**Comment by mjschultz** [comment]
shipit

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (mjschultz) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: I fixed them in https://github.com/ansible/ansible/pull/22415 which I can fix up once this is merged in.

**Comment by bobobox** [comment]
I merged @mjschultz's fixes to address @gundalow's request.

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (bobobox) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: I merged @mjschultz's fixes to address @gundalow's request.

**Comment by s-hertel** [comment]
shipit

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: This will fail on python 2.6

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/ec2_elb_facts.py:84]: Just put this on one line (or two if needed for line lengths)
```
from a.b.c import x, y, z
from a.b.c import u, v, w
```

**Review by willthames — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (bobobox) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py:84]: Is there an established Ansible style that says this should be kept on one line? (honest question, I am new to the codebase)
In general I prefer putting things like this into multiple lines as it's a lot clearer later on in diffs when elements are added or removed, and overall just easier (IMO) to read when there are more than a couple items.

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (bobobox) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: Thanks. Fixed in 189eb0f42

**Comment by bobobox** [comment]
@gundalow - could you approve your requested changes? Thanks :)

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py:84]: I'd suggest it's an undocumented preference

But it's a fairly standard way of doing things, at least in the amazon modules:

```
$ git grep  'import [^(]*,' | wc -l
75
$ git grep 'import (' | wc -l
5
```

File: lib/ansible/modules/cloud/amazon/ec2_elb_facts.py
Comment (gundalow) [reply lib/ansible/modules/cloud/amazon/ec2_elb_facts.py]: Sounds good

**Review by gundalow — APPROVED** [review]

**Review by ryansb — APPROVED** [review]

**Comment by ryansb** [comment]
Merged as https://github.com/ansible/ansible/commit/6d8edfc35ff469fbc39c52a0b5ec391aa290ae4a

**Pull request closed without merging**
